diff --git a/devtree-ops.c b/devtree-ops.c
index ab43c20..b4965de 100644
--- a/devtree-ops.c
+++ b/devtree-ops.c
@@ -51,6 +51,9 @@ static int get_devtree_address_from_parttable(struct spi_desc_t *desc, off_t *dt
     uint32_t crc = pt_header.pt_cksum.crc32;
     if (pt_header.pt_signature != FPT_SIGNATURE) return ERR_FPT_MAGIC;
     memset(&pt_header.pt_cksum, 0, sizeof(pt_header.pt_cksum));
+printf("%x %x\n", crc, cksum((unsigned char *)&pt_header, sizeof(pt_header)));
+for(int p=0;p<sizeof(pt_header);++p) printf ("%x ", ((unsigned char *)&pt_header)[p]);
+printf("\n");
     if (crc != cksum((unsigned char *)&pt_header, sizeof(pt_header))) return ERR_FPT_CKSUM;
     if (pt_header.pt_version > FPT_SUPPORTED_VERSION) return ERR_FPT_VERSION;
 
