diff -ru a/app/common/services/api-utils.js b/app/common/services/api-utils.js
--- a/app/common/services/api-utils.js	2021-09-06 19:13:16.615210559 +0300
+++ b/app/common/services/api-utils.js	2021-09-24 21:55:14.603319891 +0300
@@ -985,25 +985,25 @@
                       },
                           severityText = '', order = 0;
 
-                      if (reading.hasOwnProperty('CriticalLow') &&
+                      if (reading.hasOwnProperty('CriticalLow') && reading.CriticalLow != null &&
                           reading.Value < reading.CriticalLow) {
                         severityFlags.critical = true;
                         severityText = 'critical';
                         order = 2;
                       } else if (
-                          reading.hasOwnProperty('CriticalHigh') &&
+                          reading.hasOwnProperty('CriticalHigh') && reading.CriticalHigh != null &&
                           reading.Value > reading.CriticalHigh) {
                         severityFlags.critical = true;
                         severityText = 'critical';
                         order = 2;
                       } else if (
-                          reading.hasOwnProperty('WarningLow') &&
+                          reading.hasOwnProperty('WarningLow') && reading.WarningLow != null &&
                           reading.Value <= reading.WarningLow) {
                         severityFlags.warning = true;
                         severityText = 'warning';
                         order = 1;
                       } else if (
-                          reading.hasOwnProperty('WarningHigh') &&
+                          reading.hasOwnProperty('WarningHigh') && reading.WarningHigh != null &&
                           reading.Value >= reading.WarningHigh) {
                         severityFlags.warning = true;
                         severityText = 'warning';
@@ -1024,14 +1024,27 @@
                           content.data[key].hasOwnProperty('Unit')) {
                         severity = getSensorStatus(content.data[key]);
 
-                        if (!content.data[key].hasOwnProperty('CriticalLow')) {
-                          content.data[key].CriticalLow = '--';
-                          content.data[key].CriticalHigh = '--';
+                        content.data[key].CriticalLowEnabled  = true;
+                        content.data[key].CriticalHighEnabled = true;
+                        content.data[key].WarningLowEnabled   = true;
+                        content.data[key].WarningHighEnabled  = true;
+
+                        if (!content.data[key].hasOwnProperty('CriticalLow') || content.data[key].CriticalLow == null) {
+                          content.data[key].CriticalLow = NaN;
+                          content.data[key].CriticalLowEnabled = false;
+                        }
+                        if (!content.data[key].hasOwnProperty('CriticalHigh') || content.data[key].CriticalHigh == null) {
+                          content.data[key].CriticalHigh = NaN;
+                          content.data[key].CriticalHighEnabled = false;
                         }
 
-                        if (!content.data[key].hasOwnProperty('WarningLow')) {
-                          content.data[key].WarningLow = '--';
-                          content.data[key].WarningHigh = '--';
+                        if (!content.data[key].hasOwnProperty('WarningLow') || content.data[key].WarningLow == null) {
+                          content.data[key].WarningLow = NaN;
+                          content.data[key].WarningLowEnabled = false;
+                        }
+                        if (!content.data[key].hasOwnProperty('WarningHigh') || content.data[key].WarningHigh == null) {
+                          content.data[key].WarningHigh = NaN;
+                          content.data[key].WarningHighEnabled = false;
                         }
 
                         tempKeyParts = key.split('/');
diff -ru a/app/server-health/controllers/sensors-overview-controller.html b/app/server-health/controllers/sensors-overview-controller.html
--- a/app/server-health/controllers/sensors-overview-controller.html	2021-09-06 19:13:16.675210558 +0300
+++ b/app/server-health/controllers/sensors-overview-controller.html	2021-09-24 21:08:33.911350744 +0300
@@ -65,7 +65,7 @@
         </th>
         <td class="sensors__tbody-cell">
           <span class="sensors__tbody-cell__title">Low critical</span>
-          <span class="sensors__tbody-cell__content">
+          <span class="sensors__tbody-cell__content" ng-if="sensor.CriticalLowEnabled">
               <span class="sensors__tbody-info">{{sensor.CriticalLow}}</span>
               <span class="sensors__tbody-degree" ng-if="sensor.unit == 'C'">&deg;</span>
             <span class="sensors__tbody-unit">{{sensor.unit}}</span>
@@ -73,7 +73,7 @@
         </td>
         <td class="sensors__tbody-cell">
           <span class="sensors__tbody-cell__title">Low warning</span>
-          <span class="sensors__tbody-cell__content">
+          <span class="sensors__tbody-cell__content" ng-if="sensor.WarningLowEnabled">
               <span class="sensors__tbody-info">{{sensor.WarningLow}}</span>
               <span class="sensors__tbody-degree" ng-if="sensor.unit == 'C'">&deg;</span>
             <span class="sensors__tbody-unit">{{sensor.unit}}</span>
@@ -89,7 +89,7 @@
         </td>
         <td class="sensors__tbody-cell">
           <span class="sensors__tbody-cell__title">High warning</span>
-          <span class="sensors__tbody-cell__content">
+          <span class="sensors__tbody-cell__content" ng-if="sensor.WarningHighEnabled">
             <span class="sensors__tbody-info">{{sensor.WarningHigh}}</span>
             <span class="sensors__tbody-degree" ng-if="sensor.unit == 'C'">&deg;</span>
             <span class="sensors__tbody-unit">{{sensor.unit}}</span>
@@ -97,7 +97,7 @@
         </td>
         <td class="sensors__tbody-cell">
           <span class="sensors__tbody-cell__title">High critical</span>
-          <span class="sensors__tbody-cell__content">
+          <span class="sensors__tbody-cell__content" ng-if="sensor.CriticalHighEnabled">
             <span class="sensors__tbody-info">{{sensor.CriticalHigh}}</span>
             <span class="sensors__tbody-degree" ng-if="sensor.unit == 'C'">&deg;</span>
             <span class="sensors__tbody-unit">{{sensor.unit}}</span>
