diff -ruN a/arch/arm/dts/ast2600-reimu.dts b/arch/arm/dts/ast2600-reimu.dts
--- /dev/null
+++ b/arch/arm/dts/ast2600-reimu.dts
@@ -0,0 +1,100 @@
+/dts-v1/;
+
+#include "ast2600-u-boot.dtsi"
+
+/ {
+        model = "MCST REIMU-6564/6564m (AST2600)";
+        compatible = "mcst,reimu-6564x", "aspeed,ast2600";
+
+	memory {
+		device_type = "memory";
+		reg = <0x80000000 0x40000000>;
+	};
+
+	chosen {
+		stdout-path = &uart5;
+	};
+
+	aliases {
+		spi0 = &fmc;
+		ethernet0 = &mac0;
+		ethernet1 = &mac1;
+	};
+
+	cpus {
+		cpu@0 {
+			clock-frequency = <800000000>;
+		};
+		cpu@1 {
+			clock-frequency = <800000000>;
+		};
+	};
+};
+
+&uart5 {
+	u-boot,dm-pre-reloc;
+	status = "okay";
+};
+
+&sdrammc {
+	clock-frequency = <400000000>;
+};
+
+&wdt1 {
+	status = "okay";
+};
+
+&wdt2 {
+	status = "okay";
+};
+
+&wdt3 {
+	status = "okay";
+};
+
+&mdio {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <	&pinctrl_mdio1_default &pinctrl_mdio2_default
+			&pinctrl_mdio3_default &pinctrl_mdio4_default>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	ethphy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+
+	ethphy1: ethernet-phy@1 {
+		reg = <0>;
+	};
+};
+
+&mac0 {
+	status = "okay";
+	phy-mode = "rgmii";
+	phy-handle = <&ethphy0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii1_default &pinctrl_mac1link_default>;
+};
+
+&mac1 {
+	status = "okay";
+	phy-mode = "rgmii";
+	phy-handle = <&ethphy1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii2_default &pinctrl_mac2link_default>;
+};
+
+&fmc {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_fmcquad_default>;
+
+	flash@0 {
+		compatible = "spi-flash", "sst,w25q256";
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+	};
+};
diff -ruN a/board/aspeed/reimu_6564m/Kconfig b/board/aspeed/reimu_6564m/Kconfig
--- /dev/null
+++ b/board/aspeed/reimu_6564m/Kconfig
@@ -0,0 +1,13 @@
+if TARGET_REIMU_6564M
+
+config SYS_BOARD
+	default "reimu_6564m"
+
+config SYS_VENDOR
+	default "aspeed"
+
+config SYS_CONFIG_NAME
+	string "board configuration name"
+	default "reimu_6564m"
+
+endif
diff -ruN a/board/aspeed/reimu_6564m/Makefile b/board/aspeed/reimu_6564m/Makefile
--- /dev/null
+++ b/board/aspeed/reimu_6564m/Makefile
@@ -0,0 +1 @@
+obj-y += reimu_6564m.o
diff -ruN a/board/aspeed/reimu_6564m/reimu_6564m.c b/board/aspeed/reimu_6564m/reimu_6564m.c
--- /dev/null
+++ b/board/aspeed/reimu_6564m/reimu_6564m.c
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) ASPEED Technology Inc.
+ */
+#include <common.h>
+#include <asm/io.h>
+
+static void ast_gpio_set(char bank, int num, int val)
+{
+    #define AST_GPIO_BASE 0x1E780000
+
+    // Register 000 & 004: GPIO A,B,C,D
+    // Register 020 & 024: GPIO E,F,G,H
+    // Register 070 & 074: GPIO I,J,K,L
+    // Register 078 & 07C: GPIO M,N,O,P
+    // Register 080 & 084: GPIO Q,R,S,T
+    // Register 088 & 08C: GPIO U,V,W,X
+    // Register 1E0 & 1E4: GPIO Y,Z,{,| (AA,AB)
+    u32 regofs[] = { 0x0, 0x20, 0x70, 0x78, 0x80, 0x88, 0x1E0 };
+
+    bank = (bank | 0x20) - 'a';
+    u32 basereg = AST_GPIO_BASE + regofs[bank >> 2];
+    int bitofs  = (bank & 0x3) << 3 | num;
+
+    // Direction: output
+    writel(readl(basereg + 4) | (u32)BIT(bitofs), basereg + 4);
+
+    // Value
+    if (val) writel(readl(basereg) |  (u32)BIT(bitofs), basereg);
+    else     writel(readl(basereg) & ~(u32)BIT(bitofs), basereg);
+}
+
+int board_early_init_f(void)
+{
+	// Set BMC_BOOT# (G1) to 0
+	ast_gpio_set('g', 1, 1);
+
+	// Set BMC_ACTIVE# (O6) to 0
+	ast_gpio_set('o', 6, 1);
+
+	// Set SPI_CONNECT (O0) to 1
+	ast_gpio_set('o', 0, 1);
+
+	// Set RST_BTN# (P3) to 0
+	ast_gpio_set('p', 3, 0);
+
+	return 0;
+}
diff -ruN a/configs/ast2600_reimu_6564m_defconfig b/configs/ast2600_reimu_6564m_defconfig
--- /dev/null
+++ b/configs/ast2600_reimu_6564m_defconfig
@@ -0,0 +1,102 @@
+CONFIG_ARM=y
+CONFIG_SYS_DCACHE_OFF=y
+CONFIG_ARCH_ASPEED=y
+CONFIG_SYS_TEXT_BASE=0x0
+CONFIG_ASPEED_AST2600=y
+CONFIG_TARGET_REIMU_6564M=y
+CONFIG_SYS_MALLOC_F_LEN=0x800
+CONFIG_ENV_SIZE=0x10000
+CONFIG_ENV_OFFSET=0xE0000
+CONFIG_ARMV7_BOOT_SEC_DEFAULT=y
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_FIT=y
+CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="console=ttyS4,115200n8 root=/dev/ram rw"
+CONFIG_USE_BOOTCOMMAND=y
+CONFIG_BOOTCOMMAND="bootm 20100000"
+CONFIG_SYS_CONSOLE_ENV_OVERWRITE=y
+CONFIG_DISPLAY_BOARDINFO_LATE=y
+CONFIG_ARCH_EARLY_INIT_R=y
+CONFIG_HUSH_PARSER=y
+# CONFIG_AUTO_COMPLETE is not set
+CONFIG_SYS_PROMPT="ast# "
+CONFIG_CMD_BOOTZ=y
+# CONFIG_CMD_BOOTEFI is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_IMI is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_NVEDIT_EFI is not set
+CONFIG_CMD_MEMTEST=y
+CONFIG_SYS_ALT_MEMTEST=y
+CONFIG_CMD_CLK=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_PART=y
+CONFIG_CMD_PCI=y
+CONFIG_CMD_SF=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_DHCP=y
+CONFIG_CMD_MII=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_EXT2=y
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_EXT4_WRITE=y
+CONFIG_CMD_FAT=y
+CONFIG_CMD_FS_GENERIC=y
+CONFIG_CMD_MTDPARTS=y
+CONFIG_EFI_PARTITION=y
+CONFIG_ENV_IS_IN_SPI_FLASH=y
+CONFIG_USE_ENV_SPI_BUS=y
+CONFIG_ENV_SPI_BUS=0
+CONFIG_USE_ENV_SPI_CS=y
+CONFIG_ENV_SPI_CS=0
+CONFIG_USE_ENV_SPI_MAX_HZ=y
+CONFIG_ENV_SPI_MAX_HZ=100000000
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+CONFIG_DM_GPIO=y
+CONFIG_ASPEED_GPIO=y
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_ASPEED=y
+CONFIG_MISC=y
+CONFIG_ASPEED_AHBC=y
+CONFIG_ASPEED_H2X=y
+CONFIG_DM_MMC=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_ASPEED=y
+CONFIG_DM_SPI_FLASH=y
+CONFIG_SPI_FLASH=y
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_PHY_BROADCOM=y
+CONFIG_PHY_REALTEK=y
+CONFIG_PHY_NCSI=y
+CONFIG_DM_ETH=y
+CONFIG_PHY_GIGE=y
+CONFIG_FTGMAC100=y
+CONFIG_MDIO=y
+CONFIG_PCI=y
+CONFIG_DM_PCI=y
+CONFIG_PCIE_ASPEED=y
+CONFIG_PHY=y
+CONFIG_PINCTRL=y
+CONFIG_RAM=y
+CONFIG_DM_SERIAL=y
+CONFIG_SYS_NS16550=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_SYSRESET=y
+CONFIG_USB=y
+CONFIG_DM_USB=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_STORAGE=y
+CONFIG_WDT=y
+CONFIG_HEXDUMP=y
+# CONFIG_GENERATE_SMBIOS_TABLE is not set
+CONFIG_ASPEED_SPI_DMA=n
diff --ruN a/include/configs/reimu_6564m.h b/include/configs/reimu_6564m.h
--- /dev/null
+++ b/include/configs/reimu_6564m.h
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+/*
+ * Copyright (C) ASPEED Technology Inc.
+ */
+
+#ifndef __CONFIG_H
+#define __CONFIG_H
+
+#include <configs/aspeed-common.h>
+
+#define CONFIG_SYS_MEMTEST_START	(CONFIG_SYS_SDRAM_BASE + 0x300000)
+#define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_MEMTEST_START + 0x5000000)
+
+#define CONFIG_SYS_UBOOT_BASE		CONFIG_SYS_TEXT_BASE
+
+/* Memory Info */
+#define CONFIG_SYS_LOAD_ADDR		0x83000000
+
+#endif	/* __CONFIG_H */
diff -ruN a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -683,6 +683,7 @@ dtb-$(CONFIG_ARCH_ASPEED) += \
 	ast2600a1-ncsi.dtb \
 	ast2600-fpga.dtb \
 	ast2600-rainier.dtb \
+	ast2600-reimu.dtb \
 	ast2600-slt.dtb \
 	ast2600-tacoma.dtb
 
diff -ruN a/arch/arm/mach-aspeed/ast2600/Kconfig b/arch/arm/mach-aspeed/ast2600/Kconfig
--- a/arch/arm/mach-aspeed/ast2600/Kconfig
+++ b/arch/arm/mach-aspeed/ast2600/Kconfig
@@ -51,6 +51,12 @@
 	help
 	  SLT-AST2600 is Aspeed SLT board for AST2600 chip.
 
+config TARGET_REIMU_6564M
+	bool "REIMU-AST2600"
+	depends on ASPEED_AST2600
+	help
+	  REIMU-6564M is MCST module based on AST2600 chip.
+
 endchoice
 
 config ASPEED_SECBOOT_BL2
@@ -71,5 +77,6 @@
 source "board/aspeed/ncsi_ast2600a1/Kconfig"
 source "board/aspeed/fpga_ast2600/Kconfig"
 source "board/aspeed/slt_ast2600/Kconfig"
+source "board/aspeed/reimu_6564m/Kconfig"
 
 endif
